core/cmd/chains_commands.go

```diff
@@ -0,0 +1,225 @@
+	}
+	lower := strings.ToLower(typ)
+	article := "a"
+	if strings.ContainsAny(lower[:1], "aeiouy") {
+		article = "an"
+	}
+	return cli.Command{
+		Name:  lower,
+		Usage: fmt.Sprintf("Commands for handling %s chains", typ),
+		Subcommands: cli.Commands{
+			{
+				Name:   "create",
+				Usage:  fmt.Sprintf("Create a new %s chain", typ),
+				Action: client.CreateChain,
+				Flags:  []cli.Flag{chainID},
+			},
+			{
+				Name:   "delete",
+				Usage:  fmt.Sprintf("Delete %s %s chain", article, typ),
+				Action: client.RemoveChain,
+			},
+			{
+				Name:   "list",
+				Usage:  fmt.Sprintf("List all %s chains", typ),
+				Action: client.IndexChains,
+			},
+			{
+				Name:   "configure",
+				Usage:  fmt.Sprintf("Configure %s %s chain", article, typ),
+				Action: client.ConfigureChain,
+				Flags:  []cli.Flag{chainID},
+			},
+		},
+	}
+}
+
+// ChainClient is a generic client interface for any type of type.
+type ChainClient[C chains.Config, R presenters.ChainResource[C], P, P2 any] interface {
+	IndexChains(c *cli.Context) error
+	CreateChain(c *cli.Context) error
+	RemoveChain(c *cli.Context) error
+	ConfigureChain(c *cli.Context) error
+}
+
+type chainClient[C chains.Config, R presenters.ChainResource[C], P, P2 any] struct {
+	*Client
+	path string
+}
+
+// NewChainClient returns a new ChainClient for a particular type of chain.
+// P is a Presenter corresponding to R, and P2 is the slice variant (type P2 []P).
+func NewChainClient[C chains.Config, R presenters.ChainResource[C], P, P2 any](c *Client, name string) ChainClient[C, R, P, P2] {
+	return &chainClient[C, R, P, P2]{
+		Client: c,
+		path:   "/v2/chains/" + name,
+	}
+}
+
+// IndexChains returns all chains.
+func (cli *chainClient[C, R, P, P2]) IndexChains(c *cli.Context) (err error) {
+	var p P2
+	return cli.getPage(cli.path, c.Int("page"), &p)
+}
+
+// CreateChain adds a new  chain.
+func (cli *chainClient[C, R, P, P2]) CreateChain(c *cli.Context) (err error) {
+	chainID := c.String("id")
+	if chainID == "" {
+		return cli.errorOut(errors.New("missing chain ID [-id]"))
+	}
+	if !c.Args().Present() {
+		return cli.errorOut(errors.New("must pass in the chain config parameters [JSON blob | JSON filepath]"))
+	}
+
+	buf, err := getBufferFromJSON(c.Args().First())
+	if err != nil {
+		return cli.errorOut(err)
+	}
+
+	params := map[string]interface{}{
+		"chainID": chainID,
+		"config":  json.RawMessage(buf.Bytes()),
+	}
+
+	body, err := json.Marshal(params)
+	if err != nil {
+		return cli.errorOut(err)
+	}
+
+	var resp *http.Response
+	resp, err = cli.HTTP.Post(cli.path, bytes.NewBuffer(body))
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	defer func() {
+		if cerr := resp.Body.Close(); cerr != nil {
+			err = multierr.Append(err, cerr)
+		}
+	}()
+
+	var p P
+	return cli.renderAPIResponse(resp, &p)
+}
+
+// RemoveChain removes a specific  Chain by id.
+func (cli *chainClient[C, R, P, P2]) RemoveChain(c *cli.Context) (err error) {
+	if !c.Args().Present() {
```

Comment:
I can see the id is passed differently than for `CreateChain` and therefore the error message looks different

Reply:
Yeah, this felt a bit inconsistent. It is pre-existing though. I don't think we can switch to flag here to align since it would be breaking, but do you think the error message should be improved?

Reply:
Would be great to have the same UX (same flag) given the same semantic. And the same error message.